src/artemis/devices/oav/oav_centring.py

```diff
@@ -0,0 +1,199 @@
+
+
+class OAVCentring:
+    def __init__(self, beamline="BL03I"):
+        self.oav = OAV(name="oav", prefix=beamline + "-DI-OAV-01:")
+        self.oav_camera = Camera(name="oav-camera", prefix=beamline + "-EA-OAV-01:")
+        self.oav_backlight = Backlight(
+            name="oav-backlight", prefix=beamline + "-MO-MD2-01:"
+        )
+        self.oav_goniometer = Goniometer(name="oav-goniometer", prefix="-MO-SGON-01:")
+        self.oav_parameters = OAVParameters()
+        self.oav.wait_for_connection()
+
+    def pre_centring_setup_oav(self, context="loopCentring"):
+        """Setup OAV PVs with required values."""
+
+        self.oav_parameters.load_parameters_from_file(context)
+
+        yield from bps.abs_set(self.oav.colour_mode_pv, self.oav_parameters.rgb_mode)
+        yield from bps.abs_set(
+            self.oav.acquire_period_pv, self.oav_parameters.acquire_period
+        )
+        yield from bps.abs_set(self.oav.exposure_pv, self.oav_parameters.exposure)
+        yield from bps.abs_set(self.oav.gain_pv, self.oav_parameters.gain)
+
+        # select which blur to apply to image
+        yield from bps.abs_set(
+            self.oav.preprocess_operation_pv, self.oav_parameters.preprocess
+        )
+
+        # sets length scale for blurring
+        yield from bps.abs_set(
+            self.oav.preprocess_ksize_pv, self.oav_parameters.preprocess_K_size
+        )
+
+        # Canny edge detect
+        yield from bps.abs_set(
+            self.oav.canny_lower_threshold_pv,
+            self.oav_parameters.canny_edge_lower_threshold,
+        )
+        yield from bps.abs_set(
+            self.oav.canny_upper_threshold_pv,
+            self.oav_parameters.canny_edge_upper_threshold,
+        )
+        # "Close" morphological operation
+        yield from bps.abs_set(self.oav.close_ksize_pv, self.oav_parameters.close_ksize)
+
+        # Sample detection
+        yield from bps.abs_set(
+            self.oav.sample_detection_scan_direction_pv, self.oav_parameters.direction
+        )
+        yield from bps.abs_set(
+            self.oav.sample_detection_min_tip_height_pv,
+            self.oav_parameters.minimum_height,
+        )
+
+        # Connect MXSC output to MJPG input
+        yield from self.start_mxsc(
+            self.oav_parameters.input_plugin + "." + self.oav_parameters.mxsc_input,
+            self.oav_parameters.min_callback_time,
+            self.oav_parameters.filename,
+        )
+
+        yield from bps.abs_set(
+            self.oav.input_pv, self.oav_parameters.input_plugin + ".MXSC"
+        )
+
+        # zoom is an integer value, stored as a float in a string so
+        # we may need a .0 on the end? GDA does this with
+        # zoomString = '%1.0dx' % float(zoom)
+        # which seems suspicious, we may want to think about doing this in a nicer way
+        yield from bps.abs_set(
+            self.oav_camera.zoom, f"{float(int(self.oav_parameters.zoom))}x", wait=True
+        )
```

Comment:
Should: It's not just an integer, if you run `caget -d DBR_GR_ENUM BL03I-EA-OAV-01:FZOOM:MP:SELECT` you can get all the options for it, which include `2.5x` and `7.5x`. I think we should have a conversion function on the device itself that:
* Takes a requested zoom, compares it to the options (read from the PV)
* If the request is possible return the string to set
* If the request is not possible log a warning and return the closes zoom level

Reply:
It will need to be an integer: I'm fairly sure it currently is from I03 - if it weren't a compatible pixels to microns ration wouldn't be able to be extracted from the JSON.